### PR TITLE
partially address Issue 12939

### PR DESCRIPTION
**Pull request by WalterBright**
https://issues.dlang.org/show_bug.cgi?id=12939

### CODE REVIEW

File: src/func.d [hunk 1]
```diff
@@ -2115,7 +2115,7 @@ public:
                             bool isnothrow = f.isnothrow & !(flags & FUNCFLAGnothrowInprocess);
                             int blockexit = s.blockExit(this, isnothrow);
                             if (f.isnothrow && (global.errors != nothrowErrors))
-                                .error(loc, "%s '%s' is nothrow yet may throw", kind(), toPrettyChars());
+                                .error(loc, "nothrow %s '%s' may throw", kind(), toPrettyChars());
```
Comment (UplinkCoder) [inline src/func.d]: I liked the yet

**Comment by WalterBright** [comment]
Sheesh, this must be the most tested error message ever!

**Comment by WalterBright** [comment]
In general, templates should be inferring nothrow, pure, @nogc and @safe rather than having them specified. There are also other templates in that file with incorrect @trusted. Given the heat in the forum yesterday about @safe, there are 49 other issues I am looking at. So if someone wants to take this PR over, I'd appreciate it.

**Comment by WalterBright** [comment]
Never mind that comment, I meant it for another PR.

**Comment by andralex** [comment]
alrighty

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**